### ISSUE TITLE
Issue with INT8 TFLite Model Performance After Exporting YOLOv11m

### ISSUE THREAD

**Issue by DevinLautan** [ultralytics/ultralytics#19867]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello Ultralytics Team,​

I have fine-tuned the YOLOv11m model and achieved satisfactory results during testing with the best.pt weights. My goal is to export this fine-tuned model to the TFLite format with INT8 quantization for integration into a Kotlin-based Android application. I use Roboflow for my custom dataset.

**Export Process:**
I utilized the following code to export the model:​

```
from ultralytics import YOLO

# Load the fine-tuned model
model = YOLO('path/to/my/fine-tuned/best.pt')

# Export the model to TFLite with INT8 quantization
model.export(
    format="tflite",
    iou=0.45,
    conf=0.25,
    imgsz=640,
    half=False,
    int8=True,
    nms=False,
    batch=1,
    dynamic=False,
    data="/path/to/my/fine-tuned/data.yaml",
)
```

This process generated a saved_model directory containing several TFLite models:​
1. best_float16.tflite​
2. best_float32.tflite​
3. best_full_integer_quant.tflite​
4. best_int8.tflite​
5. best_integer_quant.tflite​

**Issue Encountered:**
When testing these models, all except best_int8.tflite produced results consistent with {Image 1} the original best.pt model. 

![Image](https://github.com/user-attachments/assets/7abc9e66-7052-40db-85a8-464dead35760)

However, using best_int8.tflite resulted in significantly degraded performance {Image 2}.​

![Image](https://github.com/user-attachments/assets/91343484-5eeb-41d7-b2e9-61a1f6b0553c)

**Request for Assistance:**
I am seeking guidance on the following:​

- Export Parameters: Are there additional parameters or configurations required during the export process to ensure optimal performance of the INT8 quantized TFLite model?​
- Calibration Dataset: Could the quality or quantity of the calibration dataset specified in the data.yaml file impact the performance of the INT8 model?​
- Known Issues: Are there any known issues or limitations with INT8 quantization for YOLOv11m models that I should be aware of?​

Despite researching existing documentation and issues, I have not found a solution to this problem. Any insights or recommendations would be greatly appreciated.​

**Comment by UltralyticsAssistant** [comment]
👋 Hello @DevinLautan, thank you for reaching out and for providing such a detailed explanation of your issue 🚀! This is an automated response to let you know that your query has been received, and an Ultralytics engineer will assist you soon.

In the meantime, here are some suggestions that might help:

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us better understand and debug the issue.

For custom training or export ❓ Questions, it is helpful to include as much detail as possible, such as training logs, dataset examples, and any configurations used during export. Additionally, consider reviewing our [Export Guide](https://docs.ultralytics.com/modes/export/) for tips and recommendations when exporting models, including TFLite-specific details.

## Upgrade

Ensure you are using the latest version of the `ultralytics` package, as updates often include fixes and improvements:

```bash
pip install -U ultralytics
```

## Resources

If you'd like to explore further, check out our [Docs](https://docs.ultralytics.com/) for comprehensive guides on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage. You might also find the [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) helpful for optimizing your workflow.  

## Community

Join the Ultralytics community to connect with other users and experts:
- [Discord](https://discord.com/invite/ultralytics) for real-time chat 🎧
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Reddit](https://reddit.com/r/Ultralytics) for collaborative threads  

## Verified Environments

Make sure you are running YOLO in a verified environment with all dependencies preinstalled. You can use:
- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: [Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## CI Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>  
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing, meaning the latest codebase has been verified across various platforms.

We appreciate your patience while we review your issue further. Stay tuned for an engineer's response! 😊

**Comment by Y-T-G** [comment]
Can you provide the output after running this command in terminal: `yolo checks`?

**Comment by DevinLautan** [comment]
@Y-T-G Here is the output:

`Ultralytics 8.3.92 🚀 Python-3.11.9 torch-2.6.0+cu124 CUDA:0 (NVIDIA GeForce RTX 3090, 24238MiB)
Setup complete ✅ (24 CPUs, 62.6 GB RAM, 905.5/937.3 GB disk)

OS                  Linux-5.15.0-134-generic-x86_64-with-glibc2.31
Environment         Linux
Python              3.11.9
Install             pip
Path                /home/nckusoc/Documents/AICup4/.venv/lib/python3.11/site-packages/ultralytics
RAM                 62.57 GB
Disk                905.5/937.3 GB
CPU                 13th Gen Intel Core(TM) i7-13700
CPU count           24
GPU                 NVIDIA GeForce RTX 3090, 24238MiB
GPU count           1
CUDA                12.4

numpy               ✅ 2.1.1<=2.1.1,>=1.23.0
matplotlib          ✅ 3.10.1>=3.3.0
opencv-python       ✅ 4.11.0.86>=4.6.0
pillow              ✅ 11.1.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.15.2>=1.4.1
torch               ✅ 2.6.0>=1.8.0
torch               ✅ 2.6.0!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.21.0>=0.9.0
tqdm                ✅ 4.67.1>=4.64.0
psutil              ✅ 7.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.14>=2.0.0`

**Comment by Y-T-G** [comment]
The `int8.tflite` inference has a known issue where the predictions are broken.

**Comment by Y-T-G** [comment]
Can you try exporting with latest Ultralytics?

And also run `pip install -U onnx2tf` before that.

**Comment by DevinLautan** [comment]
I have tried exporting with the latest Ultralytics and onnx2tf.

Current Version:
ultralytics                     8.3.96
onnx                            1.17.0
onnx_graphsurgeon    0.5.6
onnx-tf                         1.10.0
onnx2tf                        1.27.1
onnxruntime                1.21.0
onnxslim                      0.1.48
sng4onnx                     1.0.4

And it turns out to be the same, failed accuracy. It can detect, but it detect like below.

![Image](https://github.com/user-attachments/assets/32cfa37c-e36e-4d38-b07f-b3bb85a82bf2)

How can I solve this?

**Comment by DevinLautan** [comment]
I know running the conversion will first convert the .pt model to .onnx, then onnx to tflite. The exported onnx model working well during testing.

I provide the log that I get from running the conversion, hope it helps:

```
ONNX: starting export with onnx 1.17.0 opset 19...
ONNX: slimming with onnxslim 0.1.48...
ONNX: export success ✅ 1.2s, saved as '/path/to/my/best_06.onnx' (76.8 MB)
TensorFlow SavedModel: collecting INT8 calibration images from 'data=/home/nckusoc/Documents/AICup4/Combine-Dataset/data.yaml'
Scanning /path/to/my/Combine-Dataset/valid/labels.cache... 428 image
WARNING ⚠️ Box and segment counts should be equal, but got len(segments) = 233, len(boxes) = 970. To resolve this only boxes will be used and all segments will be removed. To avoid this please supply either a detect or segment dataset, not a detect-segment mixed dataset.
TensorFlow SavedModel: starting TFLite export with onnx2tf 1.27.1...
Saved artifact at '/path/to/my/best_06_saved_model'. The following endpoints are available:

* Endpoint 'serving_default'
  inputs_0 (POSITIONAL_ONLY): TensorSpec(shape=(1, 640, 640, 3), dtype=tf.float32, name='images')
Output Type:
  TensorSpec(shape=(1, 13, 8400), dtype=tf.float32, name=None)

I0000 00:00:1742954606.581844  375325 devices.cc:67] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 0
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1742954606.582151  375325 single_machine.cc:374] Starting new session
W0000 00:00:1742954608.044604  375325 tf_tfl_flatbuffer_helpers.cc:365] Ignored output_format.
W0000 00:00:1742954608.044624  375325 tf_tfl_flatbuffer_helpers.cc:368] Ignored drop_control_dependency.
I0000 00:00:1742954609.588434  375325 devices.cc:67] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 0
I0000 00:00:1742954609.588513  375325 single_machine.cc:374] Starting new session
W0000 00:00:1742954610.657061  375325 tf_tfl_flatbuffer_helpers.cc:365] Ignored output_format.
W0000 00:00:1742954610.657081  375325 tf_tfl_flatbuffer_helpers.cc:368] Ignored drop_control_dependency.
I0000 00:00:1742954611.966565  375325 devices.cc:67] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 0
I0000 00:00:1742954611.966638  375325 single_machine.cc:374] Starting new session
W0000 00:00:1742954613.034702  375325 tf_tfl_flatbuffer_helpers.cc:365] Ignored output_format.
W0000 00:00:1742954613.034722  375325 tf_tfl_flatbuffer_helpers.cc:368] Ignored drop_control_dependency.


W0000 00:00:1742954616.596681  375325 tf_tfl_flatbuffer_helpers.cc:365] Ignored output_format.
W0000 00:00:1742954616.596699  375325 tf_tfl_flatbuffer_helpers.cc:368] Ignored drop_control_dependency.
I0000 00:00:1742954616.655047  375325 mlir_graph_optimization_pass.cc:425] MLIR V1 optimization pass is not enabled
fully_quantize: 0, inference_type: 6, input_inference_type: FLOAT32, output_inference_type: FLOAT32
W0000 00:00:1742955654.502707  375325 tf_tfl_flatbuffer_helpers.cc:365] Ignored output_format.
W0000 00:00:1742955654.502725  375325 tf_tfl_flatbuffer_helpers.cc:368] Ignored drop_control_dependency.
fully_quantize: 0, inference_type: 6, input_inference_type: INT8, output_inference_type: INT8
W0000 00:00:1742956693.974770  375325 tf_tfl_flatbuffer_helpers.cc:365] Ignored output_format.
W0000 00:00:1742956693.974789  375325 tf_tfl_flatbuffer_helpers.cc:368] Ignored drop_control_dependency.
W0000 00:00:1742958887.860645  375325 tf_tfl_flatbuffer_helpers.cc:365] Ignored output_format.
W0000 00:00:1742958887.860672  375325 tf_tfl_flatbuffer_helpers.cc:368] Ignored drop_control_dependency.
TensorFlow SavedModel: export success ✅ 6594.8s, saved as '/path/to/my/best_06_saved_model' (254.6 MB)

TensorFlow Lite: starting export with tensorflow 2.19.0...
TensorFlow Lite: export success ✅ 0.0s, saved as '/home/nckusoc/Documents/AICup4/best_06_saved_model/best_06_int8.tflite' (19.6 MB)
```

**Comment by glenn-jocher** [comment]
Thank you for sharing the detailed logs. The issue appears related to the mixed detection-segmentation dataset warning during INT8 calibration. For optimal INT8 quantization:

1. Ensure your calibration dataset in `data.yaml` contains **only detection data** (no segmentation instances). The current mixed dataset causes incomplete calibration as segments are being dropped.

2. Try adding `quant_type="per-channel"` to your export command for potentially better accuracy, though it may increase inference time:

```python
model.export(..., quant_type="per-channel")
```

3. Verify your INT8 calibration images truly represent the deployment scenario's data distribution.

If issues persist, please create a minimal reproducible example following our [Bug Report Guide](https://docs.ultralytics.com/help/contributing#reproducible-example) so we can investigate further.

**Comment by DevinLautan** [comment]
Thanks for your reply, @glenn-jocher, and sorry for the late reply. However, I've tried using both suggestions, and it still didn't work. Adding the 'quant_type' into the export arguments returns "'quant_type' is not a valid YOLO argument."

I will try creating the minimal reproducible example ASAP.
Thanks.
